app/controllers/api/v2/mail_notifications_controller.rb

```diff
@@ -21,6 +22,60 @@ def index
+      param :mail_query, String, :required => false, :desc => N_("Relevant only for audit summary notification")
+
+      def create
+        @user = User.find(params[:user_id])
+        deny_access N_("You do not have permission to add a mail notification to another user") unless @user == User.current || User.current.admin?
+        mail_notification = MailNotification.find(params[:id])
+        interval_values = mail_notification.subscription_type == 'alert' ? mail_notification.subscription_options : MailNotification::INTERVALS
+        raise ::Foreman::Exception.new(N_("Interval option is not valid")) unless interval_values.include? params[:interval].capitalize
+        raise ::Foreman::Exception.new(N_("Mail query is not valid")) unless params[:mail_query].nil? || mail_notification.queryable?
+        user_mail_notification = @user.user_mail_notifications.find { |mail_notification| mail_notification.mail_notification_id == params[:id] }
+        if user_mail_notification.nil?
+          mail_notifications_params = {'interval' => params[:interval].capitalize,
+                                       'mail_notification_id' => params[:id],
+                                       'user_id' => params[:user_id],
+                                       'mail_query' => params[:mail_query]}
+          user_mail_notification = UserMailNotification.new(mail_notifications_params)
+        else
+          user_mail_notification[:interval] = params[:interval].capitalize
+          user_mail_notification[:mail_query] = params[:mail_query]
+        end
+        @user.user_mail_notifications << user_mail_notification
+        user_mail_notification
+      end
+
+      api :DELETE, "/users/:user_id/mail_notifications/:mail_notification_id", N_("Remove an email notification for a user")
+      param :user_id, String, :required => true
+      param :mail_notification_id, Integer, :required => true
+
+      def destroy
+        @user = User.find(params[:user_id])
+        if @user == User.current || User.current.admin?
+          notification = @user.mail_notifications.find(params[:id])
+          @user.mail_notifications.delete(notification)
+        else
+          deny_access N_("You do not have permission to delete a mail notification of another user")
+        end
+      end
+
+      api :GET, "/users/:user_id/mail_notifications/", N_("List all email notifications for a user")
+      param :user_id, String, :required => true
+
+      def user_mail_notifications
+        @user = User.find(params[:id])
+        @user_mail_notifications = @user.user_mail_notifications.map do |user_mail_notification|
+          mail_notification = @user.mail_notifications.find { |mail_notification| mail_notification.id == user_mail_notification.mail_notification_id }
+          user_mail_notification.attributes.merge('name' => mail_notification.name, 'description' => mail_notification.description)
+        end
+        render :json => @user_mail_notifications.as_json
+      end
```

Comment:
what about update command? if I want to change the interval and the subscription? is it possible?

Reply:
It is now implemented in the create function - both to add new mail notification and to update an existing one. I now see that it is better to add another action for an update. I will change it.